erts/emulator/beam/emu/macros.tab

```diff
@@ -74,6 +74,12 @@ GC_SWAPOUT() {
     c_p->i = I;
 }
 
+MAYBE_EXIT_AFTER_GC() {
+    if (ERTS_PROC_IS_EXITING(c_p)) {
```

Comment:
```suggestion
    if (ERTS_PSFLG_EXITING & erts_atomic32_read_nob(&c_p->state)) {
```
Avoids an unnecessary memory barrier. It might not be that big of a deal though compared to the gc if it is only used after a gc.